src/Umbraco.Web.UI.Client/src/packages/documents/documents/rollback/entity-action/rollback.action.ts

```diff
@@ -11,11 +11,16 @@ export class UmbRollbackDocumentEntityAction extends UmbEntityActionBase<never>
 		const modalManagerContext = await this.getContext(UMB_MODAL_MANAGER_CONTEXT);
 		const modalContext = modalManagerContext.open(this, UMB_ROLLBACK_MODAL, {});
 
-		await modalContext.onSubmit();
-		const notificationContext = await this.getContext(UMB_NOTIFICATION_CONTEXT);
-		notificationContext.peek('positive', {
-			data: { message: this.#localize.term('rollback_documentRolledBack') },
-		});
+		await modalContext.onSubmit().then(
```

Comment:
What is the intended difference? I think this would give the exact same outcome. :-)

Reply:
For the submit behaviour, none - the change here was to prevent the console error you would otherwise see on cancelling the dialog.  I found with just...

```
await modalContext.onSubmit().catch(() => undefined);
```

... I would get the rollback being executed even if I clicked "Cancel".  So for that reason amended to what you see now.

Reply:
ah, yes you are right about that one.

But since it is the execute() method of a extension API it should be captured by the callee, so I would think we can make it fail without `.catch(() => undefined)`.

But maybe that will first work as intended with v.16 :-/

Reply:
In which case, are you happy with how it is or would you like changes?

Reply:
It could work with the `await modalContext.onSubmit().catch(() => undefined)`, but it would require that the rollback modal (`rollback-modal.element`) provide a value (of type `UmbRollbackModalValue`).

Like so...

```ts
const data = await modalContext.onSubmit().catch(() => undefined);
if (!data) return;

// Do the notificationContext bits
```

...and in "rollback-modal.element.ts", the value would need to be set before the submit, e.g.

```
this.value = {}; // the `UmbRollbackModalValue` type has no problems, so empty object.
this.modalContext?.submit();
```

I'm happy to update this PR if we think that's a good alternative? otherwise, I'm good with the `.then()` approach.

Reply:
I'm happy with either if you are @leekelleher.  Given this is a fairly small thing, I'd suggest if you can please just update to what you see is most idiomatic with the code base and we'll go with that.

Reply:
Thanks @AndyButland, will do. 👍